pkg/collector/corechecks/cluster/ksm/kubernetes_state.go

```diff
@@ -354,78 +399,74 @@ func (k *KSMCheck) Configure(senderManager sender.SenderManager, integrationConf
 		if len(collectors) == 0 {
 			collectors = options.DefaultResources.AsSlice()
 		}
-
-		builder.WithKubeClient(apiServerClient.InformerCl)
 	}
 
-	// Prepare watched namespaces
-	namespaces := k.instance.Namespaces
+	custom := k.discoverCustomResources(client, collectors, resources)
 
-	// Enable the KSM default namespaces if the config namespaces list is empty.
-	if len(namespaces) == 0 {
-		namespaces = options.DefaultNamespaces
+	// Execute builder as a singleton, mimicking Configure()
+	// TODO: does this need to be a singleton? Is there anything unsafe in retrying?
+	if err := k.configureBuilderOnce(client, custom, collectors); err != nil {
```

Comment:
Not sure why we need the `sync.Once`, we try once anyway to call the function and if it fails we give up and do not retry it